dora/stdlib/Result.dora

```diff
@@ -26,4 +40,80 @@ impl[V, E] Result[V, E] {
       Err(value) => value,
     }
   }
+
+  @pub fun has(val: V): Bool {
+    match self {
+      Ok(actual) => actual === val,
+      Err(_) => false,
+    }
+  }
+
+  @pub fun or(alt: Result[V, E]): Result[V, E] {
+    if self.isOk() {
+      return self;
+    } else {
+      return alt;
+    }
+  }
+}
+
+impl[V: Equals, E: Equals] Result[V, E] {
```

Comment:
It doesn't look like `E: Equals` would be necessary here, right?

Reply:
Oh, well ... it would be, if I hadn't implemented `equals` below incorrectly. :-D

I'll fix that, and then the `E : Equals` makes more sense probably.